pkg/sdkserver/sdkserver_test.go

```diff
@@ -985,6 +986,186 @@ func TestSDKServerPlayerCapacity(t *testing.T) {
+		ua := action.(k8stesting.UpdateAction)
+		gs := ua.GetObject().(*agonesv1.GameServer)
+		updated <- gs.Status.Players
+		return true, gs, nil
+	})
+
+	sc.informerFactory.Start(stop)
+	assert.True(t, cache.WaitForCacheSync(stop, sc.gameServerSynced))
+
+	go func() {
+		err = sc.Run(stop)
+		assert.NoError(t, err)
+	}()
+
+	// check initial value comes through
+	// async, so check after a period
+	e := &alpha.Empty{}
+	err = wait.Poll(time.Second, 10*time.Second, func() (bool, error) {
+		count, err := sc.GetPlayerCapacity(context.Background(), e)
+		return count.Count == capacity, err
+	})
+	assert.NoError(t, err)
+
+	count, err := sc.GetPlayerCount(context.Background(), e)
+	assert.Equal(t, int64(0), count.Count)
+
+	list, err := sc.GetConnectedPlayers(context.Background(), e)
+	assert.NoError(t, err)
+	assert.Empty(t, list.List)
+
+	ok, err := sc.IsPlayerConnected(context.Background(), &alpha.PlayerID{PlayerID: "1"})
+	assert.NoError(t, err)
+	assert.False(t, ok.Bool, "no player connected yet")
+
+	// sdk value should always be correct, even if we send more than one update per second.
+	for i := 0; i < 3; i++ {
+		token := strconv.Itoa(i)
+		id := &alpha.PlayerID{PlayerID: token}
+		ok, err := sc.PlayerConnect(context.Background(), id)
+		assert.NoError(t, err)
+		assert.True(t, ok.Bool, "Player "+token+" should not yet be connected")
+
+		ok, err = sc.IsPlayerConnected(context.Background(), id)
+		assert.NoError(t, err)
+		assert.True(t, ok.Bool, "Player "+token+" should be connected")
+	}
+	count, err = sc.GetPlayerCount(context.Background(), e)
+	assert.NoError(t, err)
+	assert.Equal(t, int64(3), count.Count)
+
+	list, err = sc.GetConnectedPlayers(context.Background(), e)
+	assert.NoError(t, err)
+	assert.Equal(t, []string{"0", "1", "2"}, list.List)
+
+	// on an update, confirm that the update hits the K8s api, only once
+	select {
+	case value := <-updated:
+		assert.Equal(t, int64(3), value.Count)
+		assert.Equal(t, []string{"0", "1", "2"}, value.IDs)
+	case <-time.After(5 * time.Second):
+		assert.Fail(t, "Should have been updated")
+	}
+	agtesting.AssertEventContains(t, m.FakeRecorder.Events, "PlayerCount Set to 3")
+
+	// confirm there was only one update
+	select {
+	case <-updated:
+		assert.Fail(t, "There should be only one update for the player connections")
+	case <-time.After(2 * time.Second):
+	}
+
+	// should return an error if we try and add another, since we're at capacity
+	nopePlayer := &alpha.PlayerID{PlayerID: "nope"}
+	_, err = sc.PlayerConnect(context.Background(), nopePlayer)
+	assert.EqualError(t, err, "Players are already at capacity")
+
+	// sdk value should always be correct, even if we send more than one update per second.
+	// let's leave one player behind
+	for i := 0; i < 2; i++ {
+		token := strconv.Itoa(i)
+		id := &alpha.PlayerID{PlayerID: token}
+		ok, err := sc.PlayerDisconnect(context.Background(), id)
+		assert.NoError(t, err)
+		assert.True(t, ok.Bool, "Player "+token+" should be disconnected")
```

Comment:
Might be a bit easier to read
```suggestion
		assert.True(t, ok.Bool, "Player %s should be disconnected", token)
```